src/demo.ts

```diff
@@ -65,13 +66,13 @@ document.addEventListener('DOMContentLoaded', () => {
             cancel() {
                 log('cancel');
             },
-            purchaseCompleted(data) {
+            purchaseCompleted(data: CheckoutResponse | null) {
```

Comment:
Since you've already properly added the type to the main interface, TypeScript now properly infers it and that's the DX we want indeed.

<img width="909" alt="CleanShot 2024-12-05 at 17 18 17@2x" src="https://github.com/user-attachments/assets/12068852-ac16-403e-b7bd-4a0d4714adbc">


![CleanShot 2024-12-05 at 17 19 10@2x](https://github.com/user-attachments/assets/09cfef94-f500-4f89-8237-d9ba28da4567)

So I would suggest to rever these changes too.